recipes-examples/celebrity-face-match/demo-celebrity-face-match/demo-celebrity-face-match.service

```diff
@@ -1,16 +1,14 @@
 [Unit]
 Description=Celebrity Face Match Demo Service
-Wants=systemd-udev-settle.service weston.service
-After=systemd-udev-settle.service weston.service
+Requires=weston.service
+After=weston.service
 
 [Service]
-Type=simple
+Type=exec
 User=root
-ExecStart=bash -c ". /etc/profile ;/usr/bin/python3 /usr/share/demo-celebrity-face-match/aidemo.py"
+Environment="XDG_RUNTIME_DIR=/run/user/0 WAYLAND_DISPLAY=/run/wayland-0"
```

Comment:
I'm just curious, do you need the Environment setting when you source /etc/profile before starting the demo?

Reply:
Somehow yes, although it should be set via `/etc/profile`. Maybe only `XDG_RUNTIME_DIR` is needed here, as this is not set via `/etc/profile` if I remember correctly. I will look into it again.

Reply:
Question from me would be: shouldn't the sourcing be done in 'ExecStartPre' + shouldn't we source the user's .profile?

Reply:
The user's profile, in this case root's `.profile`, does not contain any useful things for us to source. We really want `/etc/profile` and thus the contents of `/etc/profile.d/`, as it contains variables like `WAYLAND_DISPLAY` that need to be set.

Speaking of which, we only need `XDG_RUNTIME_DIR` to be set in the service file. `WAYLAND_DISPLAY` is already set by `/etc/profile.d/weston-socket.sh`. I will remove unnecessary variables.

Using `source /etc/profile` in `ExecStartPre` does not work, as `source` is a BASH built-in command (`.` is the POSIX equivalent). You cannot use `source` or `.` outside of a shell.

Reply:
Yet, root's `.profile` does include the `/etc/profile`, does it? So imho it would be clean to use the one from the user the process runs with. True, I was wrong with the ExecStartPre suggestion.
Why not just run it within the standard root login bash: `ExecStart=/usr/bin/bash -lc /usr/bin/python3 ...` That should do the trick?!

Reply:
> Yet, root's `.profile` does include the `/etc/profile`, does it?

No, and it shouldn't. When starting BASH as a login shell, it first tries to execute `/etc/profile`. Then the shell tries to load `~/.bash_profile`, `~/.bash_login` or `~/.profile`. See `man bash` for more information. Loading profiles in the right order is the job of the used shell.

> Why not just run it within the standard root login bash: `ExecStart=/usr/bin/bash -lc /usr/bin/python3 ...` That should do the trick?!

That is a viable solution. I will use that. Thanks!